### PR TITLE
Run test(s) in context

### PR DESCRIPTION
**Pull request by teamdynamiq**
This adds a command to run test(s) depending on the context of the current cursor and open document. 

The command can be executed by right clicking the editor or executing the keyboard shortcut (default is alt-r + alt-u)

image

## Examples:

image

If the command is executed on line 10-13 it executes the dotnet test command for _Xunittest.TestClass.Pass_. If executed on line 7-9 it will run for _Xunittest.TestClass_. If it can't figure out which symbol is the matching one it will execute the test for the first symbol found in the file.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,6 +66,10 @@
           "dark": "resources/dark/run.svg"
         }
       },
+      {
+        "command": "dotnet-test-explorer.runTestInContext",
+        "title": "Run Test in context"
```
Comment (formulahendry) [inline package.json]: Better name it `Run .NET Core Test`?

File: package.json [hunk 2]
```diff
@@ -104,6 +108,12 @@
           "when": "viewItem == test",
           "group": "dotnetTestExplorer@1"
         }
+      ],
+      "editor/context": [
+        {
+          "command": "dotnet-test-explorer.runTestInContext",
+          "group": "dotnetTestExplorer@0"
```
Comment (formulahendry) [inline package.json]: This context would only be shown when language is C#, F# or VB.NET

**Comment by janaka** [comment]
Had a quick glance and seem good to me. What I mentioned in Gitter about supporting the same k/b shortcuts as Visual Studio, that would be great but I recall having conflicts with existing default shortcuts. Here they are if you can get it to work or align closer https://msdn.microsoft.com/en-us/library/da5kh0wa.aspx#bkmk_testexplorerGLOBAL

Will have a closer look at the PR later.

**Comment by stefanforsberg** [comment]
Shortcuts are tricky since what feels natural is largely what you're used to from before. If you come from vs the vs ones make sense, if you come from resharper the resharper ones make sense etc. The shortcuts are easily changeable in vscode and I have a feeling that people that care deeply about their shortcuts will change them to whatever makes sense to them regardless of what we set as a default value.

That's my gut feeling anyway.

**Pull request merged**